control/__main__.py

```diff
@@ -28,8 +30,19 @@
         help="Path to config file",
     )
     args = parser.parse_args()
+    conf = configparser.ConfigParser()
```

Comment:
@epuertat Do we need it? I mean we only have one command line argument, i.e. the config file path. I think the rest really should come from a config file. Or do you have a specific use-case in mind where command line arguments or ENV should be used instead?

Reply:
Good thing of this approach is that `argparse` is a well-known library for managing command-line arguments (and type, documentation, etc.) in a declarative way. On the contrary, `configparser` does not enforce any upfront declaration of input args.

The fact that we can also pass this args in the command-line is an free side dish, but the main goal is to type and document those:
```py
    parser.add_argument(
        "-c",
        "--config",
        default="ceph-nvmeof.conf",
        type=str,
        help="Path to config file",
    )
```
The above results in the following `--help` output (if `formatter` is set to [`ArgumentDefaultsHelpFormatter`](https://docs.python.org/3/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter), it also prints the default value):
```
# python3 -mcontrol -h
usage: python3 -m control [-h] [-c CONFIG]

Manage NVMe gateways

optional arguments:
  -h, --help            show this help message and exit
  -c CONFIG, --config CONFIG
                        Path to config file (default: ceph-nvmeof.conf)
```

However, for the config file params users have to browse through the source code and there's little information, context, defaults:
```py
tgt_path = self.config.get("spdk", "tgt_path")
```

The `pydantic` approach obviously improves this, but it's not very common to use models for input args (vs. `argparse`) and users still need to read.

Reply:
So it's more about having a automatically generated help then specifying them on the command line. That makes sense but is probably beyond the scope of this PR.

Reply:
An ability to specify all config params via command line arguments looks useful to me because it allows to have "dynamic" (built on the fly) configurations without creating a config file. I would like we had this though it may indeed be beyond the scope of this PR, and actually "nice to have" does not mean we definitely need it.